cpp/src/arrow/compute/kernels/aggregate_basic_internal.h

```diff
@@ -256,8 +262,8 @@ struct MinMaxImpl : public ScalarAggregator {
     using ScalarType = typename TypeTraits<ArrowType>::ScalarType;
 
     std::vector<std::shared_ptr<Scalar>> values;
-    if (!state.has_values ||
-        (state.has_nulls && options.null_handling == MinMaxOptions::EMIT_NULL)) {
+    if (options.min_count > 0 &&
```

Comment:
Current code returns maximal numerical limit for `min([], min_count=0)`, this is strange.
I think `min_count = 0` is not a valid case for min_max, we can verify the options and fail if it's 0.
Further more, `min_count` here is only checked for 0 or non 0, so basically it's useless if 0 is invalid. We can simply ignore it. Or improve min_max kernel (count valid values) as sum kernel does.

Reply:
This is to satisfy [expected behavior in R](https://stat.ethz.ch/R-manual/R-devel/library/base/html/Extremes.html): *The minimum and maximum of a numeric empty set are +Inf and -Inf (in this order!) which ensures transitivity, e.g., min(x1, min(x2)) == min(x1, x2).*
Perhaps it's better to fix this in R with a simple check.

Reply:
Makes sense. Thank you.